content/blog/2019/secure-egress-control-istio/index.md

```diff
@@ -0,0 +1,329 @@
+securing egress traffic is also very important. The idea is that once your cluster is compromised (and you must be
+prepared for that scenario), you want to reduce the damage as much as possible and prevent the attackers from using the
+cluster for further attacks on external services and legacy systems outside of the cluster. For that you need egress
+traffic control.
+
+Also note that you might need to control the egress traffic for compliance reasons. For example, the [Payment Card
+Industry (PCI) Data Security Standard](https://www.pcisecuritystandards.org/pci_security/) requires that both inbound
+and outbound traffic must be restricted to that which is necessary:
+
+> _1.2.1 Restrict inbound and outbound traffic to that which is necessary for the cardholder data environment, and specifically deny all other traffic._
+
+In this post I present attacks that involve egress traffic, present requirements for egress traffic control,
+describe the Istio way for controlling egress traffic securely, and compare it with alternative solutions such as
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) and legacy
+egress proxies/firewalls.
+
+Let's start with attacks that involve egress traffic.
+
+## The attacks
+
+Any IT organization must operate under the assumption that it will be attacked one day, if it is not attacked already, and
+part of its infrastructure is already compromised or will be compromised in the future.
+Once attackers are able to penetrate an application in a cluster, they can proceed to attack external services:
+legacy systems, external web services and databases. The attackers may want to steal the data of the application and to
+transfer it to their external servers. Attackers' malware may require access to attackers' servers to download
+updates. The attackers may use pods in the cluster to perform DDOS attacks or to break into external systems.
+Even though you [cannot know](https://en.wikipedia.org/wiki/There_are_known_knowns) all the possible types of
+attacks, you want to reduce possibilities for any attacks, both for known and unknown ones.
+
+The attackers can be external, that is they can gain access to the application’s container from the outside, through a
+bug in the application. The attackers can also be internal, that is, for example, malicious DevOps people inside the
+organization.
+
+To prevent the attacks described above, some form of egress traffic control must be applied. Let me present egress
+traffic control in the following section.
+
+## The solution: egress traffic control
+
+To prevent attacks involving egress traffic, all the
+egress traffic must be monitored and security policies must be enforced. You want to monitor every access to external
+services to be able to detect the attacks, possibly offline, even if you fail to prevent them in real time.
```

Comment:
@rcaballeromx The difference between ingress and egress attacks, is that egress attacks happen only when the cluster is already compromised - there are pods that attackers operate inside the cluster and these pods are attacking the external services. This is why I start describing the egress attacks with this assumption. If the assumption would be "my cluster is never compromised", then no egress attacks could happen and egress control is not required.

Reply:
Addressed in https://github.com/istio/istio.io/pull/3179/commits/9707578fb9ecf3f5ebf19c8230e30d526e73bc16.

Note that "to be able to detect the attacks" is important here. You can monitor and log all the traffic, but still do not distinguish the attacks from the legitimate traffic. You can run detection algorithms offline that can analyze the history of the requests, recognize the attacks and alert about the detected attacks.

Reply:
@rcaballeromx Oh, I see what you mean. I rewrote the sentence and connected it to the previous one: "Once you monitor the egress traffic, you can  analyze it, possibly offline, and detect the attacks, even if you were unable to prevent them in real time."

Reply:
Handled in https://github.com/istio/istio.io/pull/3179/commits/e995c10cf66f1312f3fe13958e3b9a37e0e19fc3.